packages/devtools_app/lib/src/stream_value_listenable.dart

```diff
@@ -0,0 +1,59 @@
+import 'dart:async';
+
+import 'ui/fake_flutter/fake_flutter.dart';
+
+// TODO(jacobr): remove this class and port the rest of the code in DevTools to
+// use ValueListenable directly instead of StreamBuilder.
+/// A [ChangeNotifier] to help convert stream based code to use
+/// [ValueListenable].
+class StreamValueListenable<T> extends ChangeNotifier
```

Comment:
This is a nice approach to marshalling a stream into a notifier.

Note that we may not want to actually use ChangeNotifier in situations where there are many listeners:

From https://api.flutter.dev/flutter/foundation/ChangeNotifier-class.html:

> ChangeNotifier is optimized for small numbers (one or two) of listeners. It is O(N) for adding and removing listeners and O(N²) for dispatching notifications (where N is the number of listeners).

It's about 6 lines of code to write our own notifier that is O(n) on notification dispatch.